processor/filterprocessor/filter_processor.go

```diff
@@ -65,30 +62,32 @@ func createMatcher(mp *filtermetric.MatchProperties) (*filtermetric.Matcher, err
 
 // ProcessMetrics filters the given metrics based off the filterMetricProcessor's filters.
 func (fmp *filterMetricProcessor) ProcessMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
-	mds := internaldata.MetricsToOC(md)
+	rms := md.ResourceMetrics()
 	foundMetricToKeep := false
-	for i := range mds {
-		if len(mds[i].Metrics) == 0 {
-			continue
-		}
-		keep := make([]*metricspb.Metric, 0, len(mds[i].Metrics))
-		for _, m := range mds[i].Metrics {
-			if fmp.shouldKeepMetric(m) {
-				foundMetricToKeep = true
-				keep = append(keep, m)
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ilm := ilms.At(j)
+			ms := ilm.Metrics()
+			filteredMetrics := pdata.NewMetricSlice()
+			for k := 0; k < ms.Len(); k++ {
+				metric := ms.At(k)
+				if fmp.shouldKeepMetric(metric) {
+					foundMetricToKeep = true
+					filteredMetrics.Append(metric)
+				}
 			}
+			filteredMetrics.CopyTo(ilm.Metrics())
```

Comment:
Looks like you mutate the original data and the factory returns MutatesConsumedData == false. Either copy the entire thing or change the capability to return true.

Reply:
Thanks. No longer mutating original data.